torchtune/data/_messages.py

```diff
@@ -175,30 +175,60 @@ def __init__(
     ):
         self.train_on_input = train_on_input
         self.new_system_prompt = new_system_prompt
-        if column_map:
-            if "input" not in column_map:
+
+        self.column_map = column_map
+
+        if self.column_map is not None:
+            if "input" not in self.column_map:
                 raise ValueError(
-                    f"Expected a key of 'input' in column_map but found {column_map.keys()}."
+                    f"Expected a key of 'input' in column_map but found {self.column_map.keys()}."
                 )
-            if "output" not in column_map:
+            if "output" not in self.column_map:
                 raise ValueError(
-                    f"Expected a key of 'output' in column_map but found {column_map.keys()}."
+                    f"Expected a key of 'output' in column_map but found {self.column_map.keys()}."
                 )
-            self._column_map = column_map
         else:
-            self._column_map = {"input": "input", "output": "output"}
+            self.column_map = {"input": "input", "output": "output"}
 
     def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self.column_map is not None:
+            is_multimodal = "image" in sample or (
+                "image" in self.column_map and self.column_map["image"] in sample
+            )
+        else:
+            is_multimodal = "image" in sample
+
+        if is_multimodal and "image" not in self.column_map:
+            self.column_map["image"] = "image"
+
+        if is_multimodal:
+            image_path = sample[self.column_map["image"]]
+            if isinstance(image_path, str):
+                # Load if not loaded
+                pil_image = load_image(image_path)
```

Comment:
the images themselves could be saved to some local directory if the user had to download the image dataset separately. we should provide an option to pass a path to prepend to all image paths in the dataset in this case